src/psyclone/dynamo0p3.py

```diff
@@ -2135,34 +2226,47 @@ def _invoke_declarations(self, parent):
                         datatype="reference_element_type",
                         entity_decls=[self._ref_elem_name + " => null()"]))
 
-        # Declare the necessary scalars
-        nface_vars = []
-        if self._nfaces_h_name:
-            nface_vars.append(self._nfaces_h_name)
-        if self._nfaces_v_name:
-            nface_vars.append(self._nfaces_v_name)
-
+        # Declare the necessary scalars (sort for compatibility with Python 2)
+        nface_vars = sorted(list(dict.fromkeys(self._arg_properties.values())))
```

Comment:
Because I need to remove the duplicate "nfaces_*" and this is faster and more elegant than `for` loops, list comprehensions or sets according to posts on stackoverflow. I removed sorting as the order of metadata arguments is now preserved.